### PR TITLE
fix: add imap logging and improve reliability

### PR DESCRIPTION
**Pull request by lucidNTR**
@lucychambers @jo

### CODE REVIEW

File: lib/find-emails.js [hunk 1]
```diff
@@ -75,11 +82,18 @@ module.exports = function findEmails (args, callback) {
...
     keepAlive: false,
-    // debug: function (info) { console.log(info) },
     tlsOptions: {
       secureProtocol: 'TLSv1_method'
     }
-  })
+  }
+
+  if (args.debug) {
+    imapOptions.debug = function (info) {
```
Comment (jo) [inline lib/find-emails.js:91]: why don't you use `debug` log level?

Comment (lucidNTR) [inline lib/find-emails.js:91]: how?

Comment (jo) [inline lib/find-emails.js:91]: I mean there is a logger function with different levels, eg

``` js
log('info', 'my info message')
log('debug', 'my verbose debug msg')
```

Then you can just say something like

``` js
imapOptions.debug = log.bind(null, ['debug'])
```

Comment (lucidNTR) [inline lib/find-emails.js:91]: yes sorry im stupid

Comment (jo) [inline lib/find-emails.js:91]: You're not and you know it :stuck_out_tongue:
That's exactly why we have a review process.

**Comment by lucidNTR** [comment]
hmm i think if i dont attach a handler function at all if debug is not set, its a bit cleaner

**Comment by lucidNTR** [comment]
otherwise it calls a million imap messages to the verbose logger, even the log is ignored

File: lib/find-emails.js [hunk 2]
```diff
@@ -33,28 +33,33 @@ module.exports = function findEmails (args, callback) {
...
-        args.log.error('callback called after timeout')
-      }
+    } else {
+      args.log.error('timeout hit unmet case')
+      imapDisconnect()
+    }
+
+    maybeErrorCallback({abort: true})
+
+    maybeErrorCallback = function () {
```
Comment (jo) [inline lib/find-emails.js:50]: I consider this a bit... hacky.

Comment (lucidNTR) [inline lib/find-emails.js:50]: the abort: true or the replacing of the callback? :-D

it s pretty ugly, i agree

Comment (jo) [inline lib/find-emails.js:50]: both. But more the latter.

Comment (lucidNTR) [inline lib/find-emails.js:50]: ok but that more a fallback and will be removed as soon as this is stable. it does not rely on this in any way

Comment (lucidNTR) [inline lib/find-emails.js:50]: but i will make it nicer when i replace the logger

Comment (jo) [inline lib/find-emails.js:50]: :+1:

**Comment by jo** [comment]
LGTM

```
WOW.
        such reliable
  much callback
                 so improvement
```

**Comment by lucidNTR** [comment]
:D

**Comment by jo** [comment]
Damn. I need a pre-merge hook with a confirmation window:

> Have you checked the commit messages?

https://github.com/eHealthAfrica/download-email-attachments/commit/38631110a816f1c379069c592921e4d358610d92 [Should have been a style commit](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines):

``` diff
- fix: bad whitespace 
+ style: bad whitespace 
```

**Comment by lucidNTR** [comment]
afaik semantic release only has fix and feat never saw a stlye: commit

**Comment by lucidNTR** [comment]
also if the whitespace prevents code to properly run its a fix ;-)

**Comment by lucychambers** [comment]
@lucidNTR does this relate to a specific ticket? Just if you can make sure there are tickets for everything you work on in the main VAN repo and they are referenced in the PRs I would be muchly appreciative!

**Comment by jo** [comment]
@lucidNTR if the whitespace affects code than its a fix af course.

You can use all commit semantics like `chore`, `style`, `docs` [etc](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines). But only `fix` and `feat` trigger a release.

File: lib/find-emails.js (hunk 2 above)
Comment (lucidNTR) [inline lib/find-emails.js:50]: ok improved everything but  the logger. maybe the next time i touch it

**Pull request merged**